module-services/service-db/agents/settings/FactorySettings.cpp

```diff
@@ -23,32 +23,33 @@ namespace settings
 
     void FactorySettings::initDb(Database *database)
     {
-        auto factoryData = getMfgEntries();
-        if (factoryData->getRowCount()) {
-            do {
-
-                settings::EntryPath variablePath{"",
-                                                 "",
-                                                 "",
-                                                 settings::factory::entry_key + std::string("/") +
-                                                     (*factoryData)[0].getString(),
-                                                 settings::SettingsScope::Global};
-                auto value = (*factoryData)[1].getString();
-
-                if (!database->execute(
-                        settings::Statements::insertValue, variablePath.to_string().c_str(), value.c_str())) {
-                    break;
-                }
-
-                LOG_INFO("Put factory entry: %s=%s", variablePath.to_string().c_str(), value.c_str());
-
-            } while (factoryData->nextRow());
+        const auto factoryData = getMfgEntries();
+
+        if (factoryData->getRowCount() <= 0) {
+            LOG_FATAL("No EEPROM factory data available!");
+            return;
         }
+
+        do {
+            const auto key   = (*factoryData)[0].getString();
+            const auto value = (*factoryData)[1].getCString();
+            const auto path  = settings::factory::entry_key + std::string("/") + key;
+            const settings::EntryPath variablePath{"", "", "", path, settings::SettingsScope::Global};
```

Comment:
Would it make sense to create string once?
```suggestion
            const auto variablePath = settings::EntryPath{"", "", "", path, settings::SettingsScope::Global}.to_string();
```

Reply:
Done